docs/secrets.md

```diff
@@ -253,4 +227,101 @@ azure_secret = context.get_secret("MY_AZURE_SECRET")
+code but rather is extracted from a client-side environment variable. For example, if running MLRun on a Jupyter 
+notebook and there are environment variables named `MY_SECRET` and `ANOTHER_SECRET` on Jupyter, the following code  
+passes those secrets to the executed runtime:
+
+```python
+task.with_secrets("env", "MY_SECRET, ANOTHER_SECRET")
+```
+
+When generating the runtime execution environment (for example, pod for the `job` runtime), MLRun retrieves the
+value of the environment variable and places it in the pod spec. This means that a user with `kubectl` capabilities who
+can see pod specs can still see the secret values passed in this manner.
+
+##### File
+The file provider is used to pass secret values that are stored in a local file. The file needs to be made of 
+lines, each containing a secret and its value separated by `=`. For example:
+
+```shell
+# secrets.txt
+SECRET1=123456
+SECRET2=abcdef
+```
+
+Use the following command to add these secrets:
+
+```python
+task.with_secrets("file", "/path/to/file/secrets.txt")
+```
+
+</details>
+
+## Externally managed secrets
+MLRun provides facilities to map k8s secrets that were created externally to jobs that are executed. To enable that,
+the spec of the runtime that is created should be modified by mounting secrets to it - either as files or as 
+environment variables containing specific keys from the secret.
+
+### Mapping secrets to environment
+Let's assume a k8s secret called `my-secret` was created in the same k8s namespace where MLRun is running, with two
+keys in it - `secret1` and `secret2`. The following example will add these two secret keys as environment variables
+to an MLRun job:
+
+```python
+function = mlrun.code_to_function(
+    name="secret_func",
+    handler="test_function",
+    ...
+)
+
+function.set_env_from_secret("SECRET_ENV_VAR_1", secret="my-secret", secret_key="secret1")
+function.set_env_from_secret("SECRET_ENV_VAR_2", secret="my-secret", secret_key="secret2")
+```
+
+This will only take effect for functions executed remotely, as the secret value will be injected to the function pod.
+Within the function code, the secret values will be exposed as regular environment variables, for example:
+
+```python
+# Function handler
+def test_function(context):
+    # Getting the value in the secret2 key.
+    my_secret_value = os.environ.get("SECRET_ENV_VAR_2")
+    ...
+```
+
+### Mapping secrets as files
+A k8s secret can be mapped as a filesystem folder to the function pod using the {py:func}`~mlrun.platforms.mount_secret`
+function:
+
+```python
+# Mount all keys in the secret as files under /mnt/secrets
+function.mount_secret("my-secret", "/mnt/secrets/")
+```
+
+This will create two files in the function pod, called `/mnt/secrets/secret1` and `/mnt/secrets/secret2`. Reading these
+files will provide the values. It is possible to limit the keys mounted to the function - see the documentation
```

Comment:
This creates two files in the function pod, called `/mnt/secrets/secret1` and `/mnt/secrets/secret2`. Reading these
files provides the values. It is possible to limit the keys mounted to the function - see the documentation